### PR TITLE
Confirmation is now optional.

### PR DESCRIPTION
**Pull request by eccles**
All unittests pass.

Functional tests pass with 2 changes:

1. create an asset or event without confirmation returns JSON which is not fully populated
2. confirmation is required it access to the publicurl is required immediately afterwards

[AB#9779](https://dev.azure.com/jitsuin/0629f48c-3979-4bbc-9026-cb06b3dfd0ae/_workitems/edit/9779)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -84,22 +84,34 @@ You can then use the examples code to create assets (see examples directory):
...
+        # 2. alternatively one can wait for the asset to be confirmed in the
+        #    immutable store.
+        #    The second argument is wait for confirmation:
+        #      If @confirm@ is True then this function will not
+        #      return until the asset is confirmed.
+        #
+        # Confirmation guarantees that 3rd parties can retrieve and cryptographically
+        # verify your Assets, which can take a few seconds to propagate. It is typically
+        # not necessary to wait unless your workflow involves near-real-time
+        # communication with 3rd parties and the 3rd party needs instant cryptographic
```
Comment (JAG-UK) [inline README.rst:102]:
```suggestion
        # communication with 3rd parties and the 3rd party needs immediate cryptographic
```

File: README.rst [hunk 2]
```diff
@@ -238,7 +250,12 @@ Each step consists of control parameters (specified in the 'step' dictionary) an
 the yaml representation of the request body for an asset or event.
 
 The confirm: field is a control variable for the PythonSDK that ensures that the
-asset or event is confirmed before returning.
+asset or event is confirmed before returning. This is optional and is only required
+if guarantees that 3rd parties can retrieve and cryptographically
```
Comment (JAG-UK) [inline README.rst:254]:
```suggestion
if 3rd parties need to immediately retrieve and cryptographically
```

File: examples/create_asset.py [hunk 3]
```diff
@@ -95,8 +94,14 @@ def main():
     )
 
     # Initialize connection to Archivist. max_time is the time to wait for confirmation
-    # of an asset or event creation - the default is 300 seconds but one can optionally
+    # of an asset or event creation - the default is 30 seconds but one can optionally
     # specify a different value.
+    # Confirmation guarantees that 3rd parties can retrieve and cryptographically
+    # verify your Events, which can take a few seconds to propagate. It is typically
+    # not necessary to wait unless your workflow involves near-real-time
+    # communication with 3rd parties and the 3rd party needs instant cryptographic
```
Comment (JAG-UK) [inline examples/create_asset.py:102]:
```suggestion
    # communication with 3rd parties and the 3rd party needs immediate cryptographic
```

File: examples/create_event.py [hunk 4]
```diff
@@ -76,18 +76,34 @@ def create_event(arch, asset):
...
+    # 2. alternatively one can wait for the asset to be confirmed in the
+    #    immutable store.
+    #    The second argument is wait for confirmation:
+    #      If @confirm@ is True then this function will not
+    #      return until the asset is confirmed.
+    #
+    # Confirmation guarantees that 3rd parties can retrieve and cryptographically
+    # verify your Events, which can take a few seconds to propagate. It is typically
+    # not necessary to wait unless your workflow involves near-real-time
+    # communication with 3rd parties and the 3rd party needs instant cryptographic
```
Comment (JAG-UK) [inline examples/create_event.py:94]:
```suggestion
    # communication with 3rd parties and the 3rd party needs immediate cryptographic
```

**Review by JAG-UK — CHANGES_REQUESTED** [review]

File: README.rst (hunk 1 above)
Comment (eccles) [reply README.rst]: fixed

File: README.rst (hunk 2 above)
Comment (eccles) [reply README.rst]: fixed

File: examples/create_asset.py (hunk 3 above)
Comment (eccles) [reply examples/create_asset.py]: fixed

File: examples/create_event.py (hunk 4 above)
Comment (eccles) [reply examples/create_event.py]: fixed

**Review by JAG-UK — DISMISSED** [review]
LGTM

**Review by JAG-UK — APPROVED** [review]
LEBTM

**Pull request merged**